Libraries/xcexecution/Sources/SimpleExecutor.cpp

```diff
@@ -199,8 +200,15 @@ writeAuxiliaryFiles(
                 xcformatter::Formatter::Print(_formatter->setAuxiliaryExecutable(auxiliaryFile.path()));
 
                 if (!_dryRun) {
-                    // FIXME: This should use the filesystem.
-                    if (::chmod(auxiliaryFile.path().c_str(), S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0) {
+                    Permissions permissions;
```

Comment:
This is obnoxiously verbose, and it's not the first blob like this in this PR :(

Any idea on an API to trim this?